aws/resource_aws_vpn_connection.go

```diff
@@ -245,8 +261,37 @@ func resourceAwsVpnConnection() *schema.Resource {
 func resourceAwsVpnConnectionCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
+	// Get the optional tunnel options
+	tunnel1_cidr := d.Get("tunnel1_inside_cidr").(string)
```

Comment:
These can be simplified by using `d.GetOk()` and placed where you have your other if statements 😄 

e.g.
```
if v, ok := d.GetOk("tunnel1_preshared_key"); ok {
    options[0].TunnelInsideCidr = aws.String(v.(string))
}
```